src/python/pants/backend/python/pip_requirement.py

```diff
@@ -21,8 +22,35 @@ class PipRequirement:
     """
 
     @classmethod
-    def parse(cls, line: str) -> PipRequirement:
-        return cls(pkg_resources.Requirement.parse(line))
+    def parse(cls, line: str, description_of_origin: str = "") -> PipRequirement:
+        try:
+            return cls(pkg_resources.Requirement.parse(line))
+        except Exception as e:
```

Comment:
It would be great to narrow this to the parse exception type pkg_resources raises (can't remember the exact type of the top of my head).

Reply:
Ouch, that's pretty ugly API leakage. I'm pretty sure it used to be better (they wrapped). Hrm.

Reply:
Yeah, I think that's why I went with `Exception`. But that is also ugly in its own way.